supabase_py/lib/Storage/StorageFileApi.py

```diff
@@ -0,0 +1,220 @@
+from typing import Optional
+
+import aiohttp
+import requests
+from requests import HTTPError
+import asyncio
+from supabase_py.lib.Storage.RequestError import RequestError
+
+
+class StorageFileApi():
+    """A class used to manage storage ."""
+
+    DEFAULT_SEARCH_OPTIONS = {
+        "offset": 0,
+        "sortBy": {
+            "column": 'name',
+            "order": 'asc',
+        },
+    }
+
+    DEFAULT_FILE_OPTIONS = {
+        "cacheControl": '3600',
+    }
+
+    def __init__(self, url: str, headers: dict, bucket_id: str):
+        """
+        Create a  storage  api manager
+
+        Parameters
+        ----------
+        url
+            base url for all the operation
+        headers
+            the base authentication headers
+        bucket_id
+            the id of the bucket that we want to access, you can get the list of buckets with the SupabaseStorageClient.list_buckets()
+        """
+        self.url = url
+        self.headers = headers
+        self.bucket_id = bucket_id
+        self.loop = asyncio.get_event_loop()
+
+    async def upload(self, path: str, file: str, file_options: dict = None):
```

Comment:
is there any reason this should be async - could we achieve the same thing with requests? In general I'm not sure that we should force our users to write async code

Reply:
I implemented it like this because I wanted to add a function that would allow multiple files to be loaded at the same time. I will add a sample folder (full example) to allow the user to better understand how to use the function. I needed it for the client, so I implemented it this way, but I think other users can benefit from it. we could definitely add a synchronous function.